plugwise/smile.py

```diff
@@ -334,36 +334,42 @@ async def _smile_detect_legacy(self, result, dsmrmain):
                     raise ConnectionFailedError
             else:  # pragma: no cover
                 # No cornercase, just end of the line
-                _LOGGER.error("Connected but no gateway device information found")
+                LOGGER.error(
+                    "Connected but no gateway device information found, please create \
+                     an issue on http://github.com/plugwise/python-plugwise"
+                )
                 raise ConnectionFailedError
         return model, version
 
-    async def _smile_detect(self, result, dsmrmain):
+    async def _smile_detect(self, result, dsmrmain) -> None:
         """Helper-function for connect().
         Detect which type of Smile is connected.
         """
-        model = None
+        model: str = None
```

Comment:
This can't be correct? Its typed as `str` by the value is set as `None`

Reply:
I'll type it as `str | None` as the result might be None